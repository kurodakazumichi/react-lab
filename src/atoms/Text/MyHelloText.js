import React, { Component } from 'react';

export default class MyHelloText extends Component 
{
  render() {
    return <div>こんにちは、{this.props.name}さん!</div>;
  }
}