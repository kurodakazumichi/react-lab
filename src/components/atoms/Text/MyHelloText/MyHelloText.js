import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class MyHelloText extends Component 
{
  /**
   * 型定義
   */
  static propTypes = {
    name: PropTypes.string.isRequired
  }
  
  /**
   * 規定値を定義
   */
  static defaultProps = {
    name: '名無しの権兵衛'
  };

  /** 描画 */
  render() {
    return <div>こんにちは、{this.props.name}さん!</div>;
  }
}

