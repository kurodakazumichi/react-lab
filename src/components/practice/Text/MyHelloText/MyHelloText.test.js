import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MyHelloText from './MyHelloText';

describe('<MyHelloText /> Testing', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MyHelloText />);
  })

  it('DOM', () => {
    expect(wrapper.text()).toBe('こんにちは、名無しの権兵衛さん!');
    wrapper.setProps({ name: '山田' })
    expect(wrapper.text()).toBe('こんにちは、山田さん!');
  })

  it('MyHelloText renders correctoly', () => {
    const tree = renderer
      .create(<MyHelloText />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
