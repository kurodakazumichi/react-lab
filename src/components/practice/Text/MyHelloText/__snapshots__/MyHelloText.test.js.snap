// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyHelloText /> Testing MyHelloText renders correctoly 1`] = `
<div>
  こんにちは、
  名無しの権兵衛
  さん!
</div>
`;
